misk/src/main/kotlin/misk/web/extractors/JsonBodyParameterExtractorFactory.kt

```diff
@@ -7,13 +7,19 @@ import misk.web.actions.WebAction
 import com.squareup.moshi.Moshi
 import java.util.regex.Matcher
 import javax.inject.Inject
+import javax.inject.Singleton
 import kotlin.reflect.KParameter
 import kotlin.reflect.full.findAnnotation
 import kotlin.reflect.jvm.javaType
 
+@Singleton
 internal class JsonBodyParameterExtractorFactory @Inject constructor(
     private val moshi: Moshi
 ) : ParameterExtractor.Factory {
+    /**
+     * Creates a [ParameterExtractor] that returns the JSON body of a request if the parameter
```

Comment:
I feel like these comments should go on the type, not on the method. The documentation is for the consumer of the class, not the caller of the method.

Reply:
Makes sense, when thinking about it from that perspective. Fixing.